sqlx-core/src/pool/mod.rs

```diff
@@ -490,8 +527,36 @@ impl<DB: Database> Pool<DB> {
     }
 
     /// Get the connection options for this pool
-    pub fn connect_options(&self) -> &<DB::Connection as Connection>::Options {
-        &self.0.connect_options
+    pub fn connect_options(&self) -> RwLockReadGuard<'_, <DB::Connection as Connection>::Options> {
```

Comment:
This is technically a breaking change in the interface, but `RwLockReadGuard` implements `Deref` so in practice it shouldn't break much.

Reply:
I'd rather not chance it, we've had plenty of accidental breakages already.

Reply:
The only alternative I came up with is to just return a clone of the current options.  One way or the other, the return type has to change if the inner_pool's connect_options is to become mutable.